pkg/utils/validator/clusterresourceoverride_test.go

```diff
@@ -0,0 +1,209 @@
+package validator
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	fleetv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+func TestValidateResourceSelectedByName(t *testing.T) {
+	tests := map[string]struct {
+		name       string
+		cro        fleetv1alpha1.ClusterResourceOverride
+		want       bool
+		wantErrMsg string
+	}{
+		// TODO: Add test cases.
+		"resource selected by label selector": {
+			cro: fleetv1alpha1.ClusterResourceOverride{
+				Spec: fleetv1alpha1.ClusterResourceOverrideSpec{
+					ClusterResourceSelectors: []fleetv1beta1.ClusterResourceSelector{
+						{
+							Group:   "group",
+							Version: "v1",
+							Kind:    "Kind",
+							LabelSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"key": "value",
+								},
+							},
+						},
+					},
+				},
+			},
+			want:       false,
+			wantErrMsg: "resource is not being selected by resource name",
+		},
+		"resource selected by name": {
+			cro: fleetv1alpha1.ClusterResourceOverride{
+				Spec: fleetv1alpha1.ClusterResourceOverrideSpec{
+					ClusterResourceSelectors: []fleetv1beta1.ClusterResourceSelector{
+						{
+							Group:   "group",
+							Version: "v1",
+							Kind:    "Kind",
+							Name:    "valid-name",
+						},
+					},
+				},
+			},
+			want:       true,
+			wantErrMsg: "",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := ValidateResourceSelectedByName(tt.cro); got != tt.want {
+				t.Errorf("ValidateResourceSelectedByName() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestValidateCROLimit(t *testing.T) {
+	tests := map[string]struct {
+		overrideCount int
+		want          bool
+		wantErrMsg    string
+	}{
+		// TODO: Add test cases.
+		"no overrides": {
+			overrideCount: 0,
+			want:          true,
+			wantErrMsg:    "",
+		},
+		"less than 100 overrides": {
+			overrideCount: 75,
+			want:          true,
+			wantErrMsg:    "",
+		},
+		"exactly 100 overrides": {
+			overrideCount: 100,
+			want:          false,
+			wantErrMsg:    "",
+		},
+	}
+
+	// Run the tests
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			// Create a fake client for testing
+			scheme := runtime.NewScheme()
+			if err := fleetv1alpha1.AddToScheme(scheme); err != nil {
+				t.Fatalf("failed to add scheme: %v", err)
```

Comment:
```suggestion
				t.Fatalf("Failed to add scheme: %v", err)
```

log msg starts with Uppercase